src/pmem.c

```diff
@@ -692,11 +978,36 @@ memset_nodrain_normal(void *pmemdest, int c, size_t len)
 static void *
 memset_nodrain_movnt(void *pmemdest, int c, size_t len)
 {
+	int i;
+	char buf[CHUNK_SIZE];
+	void *dest1 = pmemdest;
+	size_t cnt = 0;
+	int num_chunks;
+
 	/* way too chatty for LOG level 3 */
 	LOG(15, "pmemdest %p c 0x%x len %zu", pmemdest, c, len);
 
-	/* XXX stub version */
-	return memset(pmemdest, c, len);
+	/*
+	 * Align initial address. From there on set data in CHUNK_SIZE
+	 * bytes.
+	 */
+	memset_nodrain_normal(buf, c, CHUNK_SIZE);
```

Comment:
If *len <= CHUNK_SIZE*, then it's always better to write directly to *pmemdest*, instead of preparing data in a temporary buffer and copy it to dest using *pmem_aling/nodrain_movnt*.
In other words:
```C
if (len <= CHUNK_SIZE)
    memset_no_drain(pmemdest, c, len);
else
    ...
```